internal/app/machined/pkg/runtime/v1alpha1/bootloader/sdboot/sdboot.go

```diff
@@ -340,8 +359,10 @@ func (c *Config) install(opts options.InstallOptions) (*options.InstallResult, e
 		return nil, err
 	}
 
-	// writing UKI by version-based filename here
-	ukiPath := fmt.Sprintf("%s-%s.efi", "Talos", opts.Version)
+	// writing UKI by version-based filename + a 7 digit timestamp
+	// this is to avoid collisions with the same version UKI being installed multiple times
+	// hash is calculated from current timestamp
+	ukiPath := fmt.Sprintf("%s-%s-%07d.efi", "Talos", opts.Version, time.Now().Unix())
```

Comment:
I think it'd be nice to do the suffix only if we have a collision, and avoid it otherwise?